web/assets/show-hide-case-summary.js

```diff
@@ -0,0 +1,28 @@
+export default function showHideCaseSummary(prefix) {
+  const button = document.querySelector(
+    '[data-module="app-case-summary-toggle"]',
+  );
+  const buttonText = document.querySelector(".case-summary-toggle-text");
```

Comment:
Rather than querying `document`, it would be better to get the container for the whole case summary and look inside that. This reduces the chance that you pick up an unrelated element with a similar class.
```js
const container = button.closest('.app-caseworker-summary');
const buttonText = container.querySelector(".case-summary-toggle-text");
```

It would also be better to use custom attributes for selectors rather than classes, partly for clarity of what they're for (e.g. I was going to suggest you could remove the "case-summary" class because it's not in the CSS) but also so the browser can optimise.

So you instead might give the button text, say, `data-id="case-summary-toggle-text"` and then use:
```js
const buttonText = container.querySelector('[data-id="case-summary-toggle-text"]');
```